### PR TITLE
Forms: Add support for "block groups" in `Builder`

### PR DESCRIPTION
**Pull request by ryangjchandler**
## Description

This pull request introduces a new "block group" concept to the `Builder` field. This allows you to organise your various blocks into appropriate groups to allow easier discovery and repeated use of them.

The API for this looks something like this (right now):

```php
Builder::make('content')
	->blockGroups([
		BlockGroup::make('basic')
			->icon('heroicon-o-document-text')
			->description('Generic text and content blocks')
	])
	->blocks([
		Block::make('text')
			->group('basic')
			->schema([
				RichEditor::make('text')
					->hiddenLabel(),
			])
	])
```

- [ ] Visual changes (if any) are shown using screenshots/recordings of before and after.

> Todo: Add recording of block groups in action.

## Code style

- [ ] `composer cs` command has been run.

## Testing

- [ ] Changes have been tested.

## Documentation

- [ ] Documentation is up-to-date.

## Other

* [ ] Add `description()` support to blocks.
* [ ] Add search to the grouped block picker to allow searching for blocks and not always clicking around.
* [ ] Improve styling of group block picker
    * Icon is too small / floating in whitespace, maybe box it.
    * Separation between chosen group header and list of blocks.

### CODE REVIEW

**Comment by ryangjchandler** [comment]
This is functional, it just needs some design love from @filamentphp/design.

File: packages/forms/resources/views/components/builder/group-block-picker.blade.php [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+                    :icon-size="$group->getIconSize()"
+                    :x-on:click="$xOnClickAction"
+                >
+                    <div class="flex items-center justify-between">
+                        <div class="space-y-1">
+                            <p class="font-medium">
+                                {{ $group->getLabel() }}
+                            </p>
+
+                            @if ($description = $group->getDescription())
```
Comment (zepfietje) [inline packages/forms/resources/views/components/builder/group-block-picker.blade.php:54]: Potentially we could support description and suffix icon in the dropdown list item component instead?

Comment (ryangjchandler) [reply packages/forms/resources/views/components/builder/group-block-picker.blade.php:54]: Yeah, like I said, it's functional but needs tidy up and design love.

**Comment by danharrin** [comment]
Whoops

**Comment by danharrin** [comment]
Hey @ryangjchandler, are you planning to pick this up again?

**Comment by ryangjchandler** [comment]
@danharrin Yep, it's on my todo list!

**Comment by danharrin** [comment]
Hi Ryan, gonna close this, but feel free to restore the branch and reopen when you get around to working on it.

**Pull request closed without merging**